### PR TITLE
CI check features

### PR DESCRIPTION
**Pull request by AurevoirXavier**
To resolve issue #1136.

This is a cross verification against zepter.

- [cargo-featalign](https://github.com/hack-ink/cargo-featalign): Verifies the proper propagation of all features.
- [zepter](https://github.com/ggwpez/zepter): Checks for accidentally enabled features.

cc @ggwpez 

---
Switch to a new branch. Original PR #1537.

### CODE REVIEW

**Review by bkchr — APPROVED** [review]

**Review by sergejparity — DISMISSED** [review]

**Review by ggwpez — APPROVED** [review]
Do you know what this `true == true` is doing in the action? Or is this from some variable interpolation for debugging?

Screenshot 2023-10-02 at 14 35 52

I will try to fix the remaining missing features with Zepter since they are currently not detected.

**Comment by AurevoirXavier** [comment]
> Do you know what this `true == true` is doing in the action? Or is this after variable interpolation for some debugging?

This flag indicates that `default-features = std`. Not all projects assume that default features are std.

**Comment by AurevoirXavier** [comment]
I have decided to disable the default std features checker for this repository.

Some crates use `default-features = ["std"]`, while others do not.

For example, in https://github.com/meithecatte/enumflags2/blob/master/Cargo.toml, std is not set as its default feature. However, cargo-featalign will complain if you do not specify `default-features = false` for their crate.

**Comment by ggwpez** [comment]
Okay lets see if we got all... going to rebase

**Comment by AurevoirXavier** [comment]
@ggwpez I think it is ready now.

File: substrate/client/executor/Cargo.toml
Comment (ggwpez) [inline substrate/client/executor/Cargo.toml:61]: I actually had the same thing in Zepter until i fixed it, but it is not needed since `std` is in the `default` features for `codec` and we dont disable the default features, see https://github.com/paritytech/parity-scale-codec/blob/c781ce04706f64c9b1fffa7ed5dd5bd14ede76db/Cargo.toml#L40

Comment (AurevoirXavier) [reply substrate/client/executor/Cargo.toml:61]: Didn't notice this before. I will update this later.

Comment (AurevoirXavier) [reply substrate/client/executor/Cargo.toml:61]: I know why, actually cargo-featalign can handle this well.

**Comment by AurevoirXavier** [comment]
I made some improvements today and came across something interesting. I'll update these tomorrow.

**Comment by AurevoirXavier** [comment]
@ggwpez Found 2 problems in proc-macro crates. And it seems the `sc-exector` didn't follow the gernal std rule, so I added it to the ignore list.

File: .github/workflows/check-features.yml
Comment (chevdor) [inline .github/workflows/check-features.yml:14]:
```suggestion
        uses: hack-ink/cargo-featalign-action@bea88a864d6ca7d0c53c26f1391ce1d431dc7f34 # v0.1.1
```

**Review by chevdor — CHANGES_REQUESTED** [review]

**Review by sergejparity — APPROVED** [review]

**Comment by ggwpez** [comment]
Okay one more rebase and then we should be good to merge.

**Review by alvicsam — APPROVED** [review]

**Comment by ggwpez** [comment]
Uhg we are waiting for approval from the audit team... the new approval rules are fun 🤡

**Review by chevdor — APPROVED** [review]

**Review by louismerlin — APPROVED** [review]

**Pull request merged**
